src/posix/platform/udp.cpp

```diff
@@ -152,8 +152,16 @@ static otError transmitPacket(int aFd, uint8_t *aPayload, uint16_t aLength, cons
 #else
     msg.msg_controllen = controlLength;
 #endif
-
-    rval = sendmsg(aFd, &msg, 0);
+    for (size_t i = 0; i < OPENTHREAD_CONFIG_PLATFORM_UDP_SEND_RETRY; i++)
+    {
+        rval = sendmsg(aFd, &msg, 0);
+        if (rval > 0)
+        {
+            break;
+        }
+        otLogWarnPlat("sendmsg failed, retry: %s", strerror(errno));
```

Comment:
because this may be the last attempt.
```suggestion
        otLogWarnPlat("sendmsg(): %s", strerror(errno));
```